src/Util/ParseUrl.php

```diff
@@ -181,12 +181,12 @@ public static function getSiteinfo($url, $no_guessing = false, $do_oembed = true
 			$charset = trim(trim(trim(array_pop($matches)), ';,'));
 		}
 
-		if ($charset == '') {
-			$charset = 'utf-8';
-		}
-
-		if (($charset != '') && (strtoupper($charset) != 'UTF-8')) {
+		if ($charset && strtoupper($charset) != 'UTF-8') {
 			logger('detected charset ' . $charset, LOGGER_DEBUG);
+
+			// See https://github.com/friendica/friendica/issues/5470#issuecomment-418351211
+			$charset = str_replace('latin-1', 'latin1', $charset);
```

Comment:
You should possibly use ```str_ireplace```.

Reply:
Good idea, I'll use it when you provide me a more complete list of charset errors.